src/dataServices/dynamoDbDataService.ts

```diff
@@ -385,6 +385,30 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
         return exportJob;
     }
 
+    async getActiveSubscriptions(params: { tenantId?: string }): Promise<Record<string, any>[]> {
+        this.assertValidTenancyMode(params.tenantId);
+        const subscriptionQuery = DynamoDbParamBuilder.buildGetActiveSubscriptions(params.tenantId);
+        let queryResponse;
+        const subscriptions: Record<string, any>[] = [];
+        do {
+            // eslint-disable-next-line no-await-in-loop
+            queryResponse = await this.dynamoDb.query(subscriptionQuery).promise();
+            queryResponse.Items?.forEach((response) => {
+                const item = DynamoDBConverter.unmarshall(response);
+                subscriptions.push(item);
+            });
+            if (queryResponse.LastEvaluatedKey) {
+                // To allow subscriptionQuery.ExclusiveStartKey = queryResponse.LastEvaluatedKey,
+                // as otherwise there is an error with syntax
+                Object.defineProperty(subscriptionQuery, 'ExclusiveStartKey', {
+                    value: queryResponse.LastEvaluatedKey,
+                });
+            }
```

Comment:
you should use the correct types so that there's no need of `defineProperty` workarounds.

here `subscriptionQuery` has the type of the return type of `DynamoDbParamBuilder.buildGetActiveSubscriptions()`(specifically the type is `{ IndexName: string; TableName: string; ExpressionAttributeNames: { "#subscriptionStatus": string }; KeyConditionExpression: string; ExpressionAttributeValues: DynamoDB.AttributeMap }`) so TS does not allow to simply add the `ExclusiveStartKey` property to it.

A simple solution is to just use `const subscriptionQuery: any = ...`. It is recommended to avoid using `any` but for this short lived variable it may be ok.

A better solution would be to use a type that correctly describes a DDB query. Looks like the sdk client exposes the type `QueryInput`. Ideally functions like `DynamoDbParamBuilder.buildGetActiveSubscriptions` should use that return type.

Reply:
I updated the buildGetActiveSubscriptions to return type QueryInput, as that seemed much cleaner as you suggested!